File: services/shop-backend/src/main/java/io/miragon/shop/adapter/inbound/rest/cart/GetCartController.java
package io.miragon.shop.adapter.inbound.rest.cart;

import io.miragon.shop.application.port.inbound.CartQuery;
import io.miragon.shop.domain.shared.UserId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import static io.miragon.shop.adapter.inbound.security.SecurityConfig.USER_ID;

@RestController
@RequestMapping("/api/cart")
public class GetCartController {

    private static final Logger log = LoggerFactory.getLogger(GetCartController.class);

    private final CartQuery query;

    public GetCartController(CartQuery query) {
        this.query = query;
    }

    @GetMapping
    public ResponseEntity<CartDto> getCart() {
        var userId = new UserId(USER_ID);
        log.info("Getting cart for user: {}", userId);
        var cart = query.getCart(userId);
        return ResponseEntity.ok(CartDto.fromDomain(cart));
    }
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/adapter/inbound/rest/cart/GetCartController.kt
package io.miragon.shop.adapter.inbound.rest.cart

import io.github.oshai.kotlinlogging.KotlinLogging
import io.miragon.shop.application.port.inbound.CartQuery
import io.miragon.shop.domain.shared.UserId
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.security.oauth2.jwt.Jwt
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/cart")
class GetCartController(
    private val query: CartQuery
) {

    private val log = KotlinLogging.logger {}

    @GetMapping
    fun getCart(@AuthenticationPrincipal jwt: Jwt): ResponseEntity<CartDto?> {
        val userId = UserId(jwt.subject)
        log.info { "Getting cart for user: $userId" }
        val cart = query.getCart(userId)
        return ResponseEntity.ok(CartDto.fromDomain(cart))
    }
}